### PR TITLE
Using constant for the various field types.

### PR DESCRIPTION
**Pull request by RoySegall**
I think using the constants is much better. 

This is branch out from #187.

### CODE REVIEW

**Comment by RoySegall** [comment]
@amitaibu Ready for review.

File: src/OgGroupAudienceHelper.php [hunk 1]
```diff
@@ -24,6 +24,15 @@ class OgGroupAudienceHelper {
    */
   const DEFAULT_FIELD = 'og_group_ref';
 
+  /**
+   * The name of the field which reference user to groups.
+   */
+  const USER_REFERENCE_FIELD = 'og_membership_reference';
```
Comment (pfrenssen) [inline src/OgGroupAudienceHelper.php:30]: This is not directly related to this PR, but why do we still use this field? We now have the OgMembership entity which tracks which groups a user is a member of. We don't need this field any more.

Comment (pfrenssen) [inline src/OgGroupAudienceHelper.php:30]: Typically when defining multiple variations of related constants, the 'fixed' part is put in front, so this is easier to find with autocompletion in the IDE. So it would be better to call it `REFERENCE_FIELD_USER`.

Some examples from core:
- `SAVED_NEW`, `SAVED_UPDATED`, `SAVED_DELETED`, ...
- `ERROR_REPORTING_HIDE`, `ERROR_REPORTING_DISPLAY_ALL`, ...
- `TRANSLATION_REMOVED`, `TRANSLATION_EXISTING`, `TRANSLATION_CREATED`, ...
- etc etc

File: src/OgGroupAudienceHelper.php [hunk 2]
```diff
@@ -24,6 +24,15 @@ class OgGroupAudienceHelper {
...
+
+  /**
+   * The name of the field which reference non-user entities to groups.
+   */
+  const NON_USER_REFERENCE_FIELD = 'og_standard_reference';
```
Comment (pfrenssen) [inline src/OgGroupAudienceHelper.php:35]: It's a bit weird to call this the "non-user" reference field. This is after all the default one, calling it non-user makes it sound like it is some kind of an exception. Maybe `REFERENCE_FIELD_STANDARD` would be better?

Again, we should actually get rid of the user referencing field, then this can become simply `OG_REFERENCE_FIELD`.

**Comment by pfrenssen** [comment]
I created an issue to remove the og membership reference field: https://github.com/amitaibu/og/issues/195.

File: src/OgGroupAudienceHelper.php (hunk 1 above)
Comment (RoySegall) [inline src/OgGroupAudienceHelper.php:30]: We still need this field because we still need a "field" on the user object to know to which groups the user belong to.

**Comment by RoySegall** [comment]
@amitaibu @pfrenssen What about the names `OG_REFERENCE_FIELD` and `OG_USER_REFERENCE_FIELD`?

File: src/OgGroupAudienceHelper.php (hunk 2 above)
Comment (amitaibu) [inline src/OgGroupAudienceHelper.php:35]: I agree the names should be changed, but for different reasons :)

We do need to have the fields - as they are acting as the reference between the entity and the group.

However we have 2 different fields based on the host entity (i.e. user or non-user).

`USER_REFERENCE_FIELD` sounds as if it is referencing a user, but in fact it just indicates it is _attached_ to a user.

so maybe it would be more explicit to name them:
- `GROUP_REFERENCE_FIELD_ON_USER`
- `GROUP_REFERENCE_FIELD_ON_NON_USER`

Comment (amitaibu) [inline src/OgGroupAudienceHelper.php:35]: @pfrenssen what do you think about the suggested names? ^^

**Comment by pfrenssen** [comment]
I'm fine with this, I was just pointing out how this is typically done in core. It's not a rule that is set in stone.

The important thing is that the constants are there so we can use them :)

+1 from me

**Comment by amitaibu** [comment]
I'll re-open a new PR against `8.x-1.x`

**Comment by amitaibu** [comment]
I'll cherry pick the changes, as the branch holds wrong stuff now.
